kartothek/io_components/metapartition.py

```diff
@@ -1383,10 +1409,19 @@ def merge_metapartitions(metapartitions, label_merger=None, metadata_merger=None
         LOGGER.debug("Merging metapartitions")
         data = defaultdict(list)
         new_metadata_version = -1
+        logical_conjunction = None
+
         for mp in metapartitions:
             new_metadata_version = max(new_metadata_version, mp.metadata_version)
             for label, df in mp.data.items():
                 data[label].append(df)
+            if mp.logical_conjunction or logical_conjunction:
```

Comment:
This `or` appears to be unused as you have just set `logical_conjunction = None` a few lines back

Reply:
You can probably simplify the if-else statement below also

Reply:
To be honest, I do not understand what this should do functionally. Could you maybe clarify @fjetter ? We iterate over the `metapartitions` and determine the value of `logical_conjunction` within this iteration -- this would be the last not-None value of `mp.logical_conjunction`. Or, if `mp.logical_conjunction` is `None` for all `mp`, then `logical_conjunction` would be `None`?

Reply:
It is not unused since the instance attribute and the variable are not the same. The variable is used to track the unique values and yes, there are many ways to do so. Since the logical_conjunction is not a trivial data type (it's a list) the straight forward approaches, e.g. using a set, doesn't work w/out manipulation of the values.
Since this is a matter of style, I don't see a reason why this should block a merge

Reply:
You're right. Should've looked into it more before requesting change